ledger/participant-state/kvutils/src/main/scala/com/daml/ledger/participant/state/kvutils/CommitPackageValidationMode.scala

```diff
@@ -0,0 +1,29 @@
+// Copyright (c) 2020 Digital Asset (Switzerland) GmbH and/or its affiliates. All rights reserved.
+// SPDX-License-Identifier: Apache-2.0
+
+package com.daml.ledger.participant.state.kvutils
+
+// define the different package validation modes.
+sealed abstract class CommitPackageValidationMode extends Product with Serializable
+
+object CommitPackageValidationMode {
+
+  // Specify that the package committer should validate the packages
+  // before committing them to the ledger.
+  case object PreCommit extends CommitPackageValidationMode
+
+  // Specify that the engine should validate the packages every time
+  // it loads them from the ledger.
+  // This mode is useful for ledger integrations that cannot handle
+  // long-running submissions (> 10s).
+  // See PackageCommitter for more detail.
+  // FIXME: Currently `PostCommit` is the mode used by default
```

Comment:
We should have an issue and a plan for addressing this. Probably not worth leaving a FIXME in the code.

Reply:
I opened KVL-578 to make the validation mode configurable.